### DISCUSSION TITLE
MagnumPlugins libraries have no lib prefix

### DISCUSSION THREAD

**Discussion by Desperado17** [mosra/magnum-plugins#144] [Q&A] [answered]
Greetings,

it seems to me that magnum plugins libraries are not generated with lib prefix: FreeTypeFont.so, AssimpImporter.so etc.

This causes problems with out build system which expects these to have lib prefixes.
Is this wanted and if yes is there are quick way to change the output format?

Regards

**Comment by mosra** [comment, marked as answer]
That's intentional, these are all put to subdirectories based on the plugin interface and without a lib prefix, as they're meant to be loaded dynamically at runtime using the plugin manager. 

- [Here's an introductionary doc chapter for the plugin manager](https://doc.magnum.graphics/magnum/plugins.html)
- [Here's a small self-contained example that's loading and using the TGA importer plugin](https://doc.magnum.graphics/magnum/examples-texturedquad.html)

Linking and using dynamic plugins directly as if they were shared libraries is not supported, if you don't want to perform dynamic loading at runtime, you can build the library and the plugins as static, but you'll lose quite a bit of flexibility that way, and the dependency libraries will be loaded in memory even though given plugin never gets actually used.
